### PR TITLE
PLFM-8005: Added stack parameter to original version in orginfra

### PR DESCRIPTION
**Pull request by xschildw**
This PR is a parameterized version of https://github.com/Sage-Bionetworks-IT/organizations-infra/blob/master/sceptre/synapseprod/templates/snowflake-access.yaml (added 'Stack' parameter).

### CODE REVIEW

**Review by zaro0508 — CHANGES_REQUESTED** [review]
if the file in this PR serves the same purpose as the existing `snowflake-access.yaml` file  then why do you need to create a new file?  Can't you just update the existing file?

**Comment by xschildw** [comment]
I'm moving it from sceptre/synapseprod/templates to aws-infra (and adding a param to support the stack) and will be using it for both synapseprod and synapsedev (synapsedev has a hardcoded test stack that is not in orginfra, we'll delete it and move everything to org-infra).

**Review by marcomarasca — COMMENTED** [review]
Shouldn't we delete the existing snowflake yaml template in the synapse porod?

**Comment by xschildw** [comment]
@marcomarasca , correct. The original file will be deleted from the organization-infra repo.

File: templates/IAM/snowflake-synapse-access.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          -
+            Effect: Allow
+            Action:
+                - s3:ListBucket
+                - s3:GetBucketLocation
+            Resource: !Sub "arn:aws:s3:::${Stack}.datawarehouse.sagebase.org"
+            Condition:
+              StringLike":
+                "s3:prefix":
+                  - "warehouse/*"
```
Comment (zaro0508) [inline templates/IAM/snowflake-synapse-access.yaml:37]:
```
            Condition:
              StringLike":
                "s3:prefix":
                  - !Sub "${BucketPrefix}/*"
```

File: templates/IAM/snowflake-synapse-access.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            Action:
+              - s3:GetObject
+              - s3:GetObjectVersion
+            Resource: !Sub "arn:aws:s3:::${Stack}.datawarehouse.sagebase.org/warehouse/*"
+          -
+            Effect: Allow
+            Action:
+                - s3:ListBucket
+                - s3:GetBucketLocation
+            Resource: !Sub "arn:aws:s3:::${Stack}.datawarehouse.sagebase.org"
```
Comment (zaro0508) [inline templates/IAM/snowflake-synapse-access.yaml:33]: to make it more generic i suggest you pass in a bucket ARN and prefix as parameters.
example:

```
Parameters:
  BucketArn:
    Type: String
  BucketPrefix:
    Type: String
```

for the PolicyDocument..
```
      PolicyDocument:
        Version: 2012-10-17
        Statement:
          -
            Effect: Allow
            Action:
              - s3:GetObject
              - s3:GetObjectVersion
            Resource: !Sub "${BucketArn}/${BucketPrefix}/*"
          -
            Effect: Allow
            Action:
                - s3:ListBucket
                - s3:GetBucketLocation
            Resource: !Sub "${BucketArn}"
```

**Review by zaro0508 — CHANGES_REQUESTED** [review]
Ahh, i see. sorry i missed that the existing file is not in aws-infra.

File: templates/IAM/snowflake-synapse-access.yaml (hunk 2 above)
Comment (xschildw) [reply templates/IAM/snowflake-synapse-access.yaml]: Great idea! Renamed since not tied to Synapse anymore. Thx!

**Review by zaro0508 — APPROVED** [review]

**Review by marcomarasca — APPROVED** [review]

**Pull request merged**